amespahdbpythonsuite/mcfitted.py

```diff
@@ -0,0 +1,306 @@
+                      f'{np.mean(results[key]):.5f} '
+                      f'{np.std(results[key]):.5f}')
+
+    def averagespec(self, **keywords):
+        """
+        Calculate average and std spectra.
+
+        """
+        #  Getting the average spectra
+        components = {}
+        lst_classes = {}
+        avg_classes = {}
+        std_classes = {}
+
+        # Get average and std spectrum.
+        if 'fits' not in keywords:
+            fits = self.getfit()
+        avg_fit = np.mean(fits, axis=0)
+        std_fit = np.std(fits, axis=0)
+
+        components['fit'] = {'mean_spec': avg_fit, 'std_spec': std_fit}
+
+        # Get average and std spectrum of classes.
+        if 'classes' not in keywords:
+            classes = self.getclasses()
+        classkeys = list(classes[0].keys())
+
+        for key in classkeys:
+            lst_classes[key] = []
+            avg_classes[key] = []
+            std_classes[key] = []
+        for c in classes:
+            for key in classkeys:
+                if key in c.keys():
+                    lst_classes[key].append(c[key])
+        for key in classkeys:
+            avg_classes[key] = np.mean(lst_classes[key], axis=0)
+            std_classes[key] = np.std(lst_classes[key], axis=0)
+            components[key] = {'mean_spec': avg_classes[key], 'std_spec': std_classes[key]}
+
+        return components
+
+    def getfit(self, **keywords) -> list:
+        """
+        Retrieves the sum of fit values.
+
+        """
+        fits = []
+        for fit in self.mcfits:
+            fits.append(fit.getfit())
+
+        return fits
+
+    def getbreakdown(self, **keywords) -> dict:
+        """
+        Retrieves the breakdown of the MC fitted PAHs.
+
+        """
+        results = dict()  # type: dict
+        results = {'solo': [],
+                   'duo': [],
+                   'trio': [],
+                   'quartet': [],
+                   'quintet': [],
+                   'anion': [],
+                   'neutral': [],
+                   'cation': [],
+                   'small': [],
+                   'large': [],
+                   'nitrogen': [],
+                   'pure': [],
+                   'nc': [],
+                   'error': []
+                   }
+
+        keys = list(results.keys())
+
+        # Obtain fit breakdown.
+        for fit in self.mcfits:
+            bd = fit.getbreakdown()
+            for key in keys[:-1]:
+                results[key].append(bd[key])
+            # Obtain fit uncertainty.
+            error = fit.geterror()['err']
+            results['error'].append(error)
+
+        return results
+
+    def getclasses(self, **keywords) -> list:
+        """
+        Retrieves the spectra of the different classes of the MC fitted PAHs.
+
+        """
+        self.mcclasses = []
+        for fit in self.mcfits:
+            self.mcclasses.append(fit.getclasses())
+
+        return self.mcclasses
+
+    def plot(self, **keywords):
```

Comment:
As part of a future 'Code Crunch/Sprint' we'll need to refactor/clean-up the plotting.